ibis/expr/operations/core.py

```diff
@@ -33,6 +33,11 @@ def equals(self, other) -> bool:
     # def __children__(self):
     #     return super().__children__
 
+    @attribute
+    def mass(self):
+        """Set of relations the value node depends on."""
+        return 1 + sum(n.mass for n in self.__children__)
```

Comment:
Since you're introducing a heuristic here instead of disabling the optimization altogether, it needs some rationale in the docstring.

- How did you arrive at this particular calculation?
- Where does it breakdown?
- How does it address the problem that we observed with inlining?